pkg/local/sync.go

```diff
@@ -304,18 +305,35 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 	metadata := map[string]string{
 		ClientMtimeMetadataKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
 	}
-	reader := fileWrapper{
+
+	reader := b.Reader(f)
+	if s.includePerm {
+		if strings.HasSuffix(path, uri.PathSeparator) { // Create a 0 byte reader for directories
```

Comment:
(Nit, I guess:) I think that extracting this block into a function (`wrapReaderWithPerms()`?) would've make this more readable.

Reply:
Usually I would agree, but see my previous comment